lib/Tools/LinearLayout.cpp

```diff
@@ -313,6 +313,10 @@ LinearLayout::checkInvariants(bool requireSurjective) {
            "can be reached by some `in` coordinate, but was not:" +
            toString();
   }
+
+  this->injective =
+      this->surjective && getTotalInDimSize() == getTotalOutDimSize();
```

Comment:
Note this is the definition of `invertible`, not `injective`.
Also, no need to cache it I don't think.